src/components/BookDetails.js

```diff
@@ -1,27 +1,64 @@
 import React, { Component } from "react";
 import { apiCalls } from "../apiCalls";
 import '../styles/BookDetails.css';
-import { cleanBookData } from "../utils";
+import { cleanBookData, trimBookData } from "../utils";
+
 
 class BookDetails extends Component {
   constructor(props) {
     super(props);
     this.state = {
       selectedBook: {},
-      error: ''
+      error: '',
     }
   }
 
   componentDidMount = () => {
     apiCalls.getSingleBook(this.props.isbn)
       .then(bookData => {
         const cleanedBookData = cleanBookData(bookData[0]);
-        this.setState({ selectedBook: cleanedBookData });
+        this.setState({ selectedBook: cleanedBookData});
       })
       .catch(error => this.setState({ error: error.message }))
   }
 
+  handleDelete = () => {
+    Promise.all([apiCalls.deleteFromFavorites(this.state.selectedBook.isbn), apiCalls.updateFavStatus(this.state.selectedBook)])
+    .then(() => {
+      return apiCalls.getSingleBook(this.props.isbn)
+        .then(data => {
+          const cleanedBookData = cleanBookData(data[0])
+          console.log('delete from fav', cleanedBookData)
+          this.setState({ selectedBook: cleanedBookData })
+        })
+    })
+    .catch(error => this.setState({ error: error.message }))
+  }
+
+  handleAdd = () => {
+    Promise.all([apiCalls.addToFavorites(trimBookData(this.state.selectedBook)), apiCalls.updateFavStatus(this.state.selectedBook)])
+    .then(() => {
+      return apiCalls.getSingleBook(this.props.isbn)
+        .then(data => {
+          const cleanedBookData = cleanBookData(data[0])
+          console.log('add to fav', cleanedBookData)
+          this.setState({ selectedBook: cleanedBookData })
+        })
+    })
+    .catch(error => this.setState({ error: error.message }))
+  }
+
+  determineButton = () => {
```

Comment:
This is a great helper function to determine what the button is supposed to do/show